### ISSUE TITLE
Unable to scan for next token. Stuck on '

### ISSUE THREAD

**Issue by felipecds** [kthompson/glob#43]
I'm using a .gitignore file, and I'm stuck on this error. The line on .gitignore file contains the value Generated\ Files/

Here are the code I'm using:

```
var directoryInfo = new DirectoryInfo(dir);
foreach (var line in File.ReadAllLines($@"{dir}\.gitignore", Encoding.UTF8))
                {
                    try
                    {
                        var cleanLine = line.Trim();

                        if (string.IsNullOrEmpty(cleanLine) || cleanLine.Length == 0) continue; //Empty line
                        if (cleanLine.Substring(0, 1) == "#") continue;
                        if (cleanLine.Substring(0, 1) == "!")
                        {
                            cleanLine = cleanLine.Substring(1);

                            var infoAll = directoryInfo.GlobFileSystemInfos("**/*");
                            var infoIgnore = directoryInfo.GlobFileSystemInfos($"**/{cleanLine}");

                            ignore.AddRange(infoAll.Except(infoIgnore));
                        }
                        else
                        {
                            ignore.AddRange(directoryInfo.GlobFileSystemInfos($"**/{cleanLine}"));
                        }
                    }
                    catch (Exception e) { continue; };
                }
```

and the .gitignore file can be obtained [here](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore)

**Comment by felipecds** [comment]
Here are the full stack of the Exception:

System.Exception: Unable to scan for next token. Stuck on ' '\r\n   at GlobExpressions.Scanner.ScanToken()\r\n   at GlobExpressions.Scanner.Scan()\r\n   at GlobExpressions.Parser.AcceptIt()\r\n   at GlobExpressions.Parser.ParseTree()\r\n   at GlobExpressions.PathTraverser.Traverse(DirectoryInfo root, String pattern, Boolean caseSensitive, Boolean emitFiles, Boolean emitDirectories)\r\n   at GlobExpressions.Glob.FilesAndDirectories(DirectoryInfo workingDirectory, String pattern, GlobOptions options)\r\n   at IgnoreBackup.Program.Main(String[] args)

The lines with errors on .gitignore file are 36, 115 and 218

**Comment by kthompson** [comment]
Looks like its getting stuck on the `$` for `$tf/` and `~$*` and possibly getting stuck on the `\` in `Generated\ Files/`  escaping is not currently supported but it is in the works(https://github.com/kthompson/glob/issues/41).

I will take a look at adding support for `$` in paths.

**Comment by felipecds** [comment]
Thanks a lot, and take your time for doing it.  I'm using Glob to make a backup tool using gitignore files for cleaning things.

**Comment by kthompson** [comment]
A prerelease version is available here:

https://www.nuget.org/packages/Glob/1.1.2-CI-20190306-040219

Please let me know if it fixes your issue.

**Comment by kthompson** [comment]
This should be closed by 1.1.2 please let me know if you run into issues.
